pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -123,37 +132,45 @@ type FeatureFlags struct {
 type Options struct {
 	Client kubeclient.Client
 
-	Revision              string
-	SystemNamespace       string
-	DomainSuffix          string
-	ClusterID             cluster.ID
-	XDSUpdater            model.XDSUpdater
-	LookupNetwork         LookupNetwork
-	LookupNetworkGateways LookupNetworkGateways
-	StatusNotifier        *activenotifier.ActiveNotifier
-	Flags                 FeatureFlags
+	Revision        string
+	SystemNamespace string
+	DomainSuffix    string
+	ClusterID       cluster.ID
+	IsConfigCluster bool
+	XDSUpdater      model.XDSUpdater
+	StatusNotifier  *activenotifier.ActiveNotifier
+	Flags           FeatureFlags
 
-	MeshConfig krt.Singleton[MeshConfig]
+	MeshConfig meshwatcher.WatcherCollection
 
-	Debugger *krt.DebugHandler
+	Debugger                    *krt.DebugHandler
+	ClientBuilder               multicluster.ClientBuilder
+	RemoteClientConfigOverrides []func(*rest.Config)
 }
 
 func New(options Options) Index {
```

Comment:
This is called within controller.NewController, which is constructed for each cluster, so by constructing ambientIndex for all the clusters here, doesnot  itmean that this is duplicate?

Reply:
Yes ambientindex is currently duplicated for each cluster, but we only create the multicluster index for config clusters. In the PR on top of this one, I think we'll only create the ambient index for config clusters